snaps/build-and-release-snaps-on-new-release.sh

```diff
@@ -6,147 +6,183 @@ set -eux
 source utilities.sh
 
 # K8s versions we want to check for new releases
-declare -a VERSIONS=('1.6' '1.7' '1.8' '1.9' '1.10' '1.11' '1.12' '1.13' '1.14' '2.0' 'latest');
+declare -a VERSIONS=('latest' '1.6' '1.7' '1.8' '1.9' '1.10' '1.11' '1.12' '1.13' '1.14' '2.0');
 
 SNAP_INFO_KUBECTL=$(snap info kubectl)
-SNAPCRAFT_REVISIONS_KUBELET=$(snapcraft revisions --arch=amd64 kubelet)
 
-function check_for_release {
-  # Sets $trigger to 'yes' when a new release is available.
-  # Compares snap info on kubectl to find version information
-
-  if [ "$BRANCH" == "latest" ]
+function is_upstream_newer_than_edge {
+  # Sets $upstream_is_newer to 'yes' when there is a version upstream newer to what we have on edge.
+  # Uses snap info on kubectl to find the snap version information
+  #
+  # Input:
+  # $TRACK eg "latest" or "1.11" to check for release
+  # $KUBE_VERSION k8s version available upstream for $TRACK
+  # $SNAP_INFO_KUBECTL snap command to get the info of kubectl
+  #
+  # Output:
+  # $upstream_is_newer 'yes' when a release upstream is newer that the snap on edge
+
```

Comment:
I love all these comments. They really clear up intentions and inputs. I understand now what you were looking for in Portland.